include/functional/detail/not_tuple.hpp

```diff
@@ -8,47 +8,29 @@
 
 namespace fn::detail {
 
-template <typename... Args> struct not_tuple;
+template <std::size_t I, typename T> struct _element {
+  T v;
 
-template <typename Arg0> struct not_tuple<Arg0> final {
-  constexpr static unsigned size = 1;
-  Arg0 v0;
-
-  static_assert(!std::is_rvalue_reference_v<Arg0>);
+  static_assert(not std::is_rvalue_reference_v<T>);
 };
 
-template <typename Arg0, typename Arg1> struct not_tuple<Arg0, Arg1> final {
-  constexpr static unsigned size = 2;
-  Arg0 v0;
-  Arg1 v1;
-
-  static_assert(!std::is_rvalue_reference_v<Arg0>
-                && !std::is_rvalue_reference_v<Arg1>);
+template <typename, typename...> struct not_tuple_base;
```

Comment:
could we rename not_tuple to tupleoid or something? "not tuple" is ... just ... blech

Reply:
how about `closure` ?